src/main.ts

```diff
@@ -152,11 +155,19 @@ const createWindow = async () => {
   }
 
   mainWindow.once('ready-to-show', () => configureWatcher(mainWindow))
+  proxyEmitter.on('status:change', (statusName: ProxyStatus) => {
+    proxyStatus = statusName
+    mainWindow.webContents.send('proxy:status:change', statusName)
+  })
```

Comment:
Since we already have an EventEmitter for the proxy I used it as a wrapper for changes related to the proxy status.